src/hinge.cpp

```diff
@@ -0,0 +1,135 @@
+#include "wayfire/plugin.hpp"
+#include "wayfire/signal-definitions.hpp"
+#include "wayfire/input-device.hpp"
+#include <wayfire/util/log.hpp>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <unistd.h>
+#include <thread>
+
+class wayfire_hinge : public wf::plugin_interface_t
+{
+    enum thread_message: char {
+        ENABLE_INPUT,
+        DISABLE_INPUT,
+        THREAD_EXIT
+    };
+
+    wf::option_wrapper_t<std::string> file_name{"hinge/filename"};
+    wf::option_wrapper_t<int> poll_freq{"hinge/poll_freq"};
+    wf::option_wrapper_t<int> flip_degree{"hinge/flip_degree"};
+    
+    int pipefd[2];
+    std::thread thread;
+    wl_event_source* pipe_reader;
+    bool exiting = false;
+
+    std::vector<nonstd::observer_ptr<wf::input_device_t>> get_inputs() {
+        wf::compositor_core_t* core = &wf::get_core();
+        return core->get_input_devices();
+    }
+
+    void disable_inputs() {
+        for (auto inp: get_inputs())
+        {
+            auto inp_type = inp->get_wlr_handle()->type;
+            if(
+            inp_type == wlr_input_device_type::WLR_INPUT_DEVICE_KEYBOARD || 
+            inp_type == wlr_input_device_type::WLR_INPUT_DEVICE_POINTER
+            ) {
+                inp->set_enabled(false);
+            }
+        }
+    }
+
+    void enable_inputs() {
+        for (auto inp: get_inputs())
+        {
+            inp->set_enabled();
+        }
+    }
+
+    static void setup_thread(std::string fn, int poll_freq, int flip_degree, bool* exiting, int pipe) {
```

Comment:
Maybe it's a good idea to rename `int pipe` to `int fd` or something that doesn't shadow the function name `pipe()` and make it clearer what is described.

Reply:
I renamed it to write_fd to be even clearer.